src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -157,6 +157,20 @@ public function setContentDisposition($disposition, $filename = '', $filenameFal
             $filename = $this->file->getFilename();
         }
 
+        if ('' === $filenameFallback && (!preg_match('/^[\x20-\x7e]*$/', $filename) || false !== strpos($filename, '%'))) {
```

Comment:
The thing is that people will expect the `ResponseHeaderBag::makeDisposition()` method to throw an exception when invalid characters are passed. Silently convertin the fallback filename is a BC break imo.